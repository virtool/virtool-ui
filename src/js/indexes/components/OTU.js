import PropTypes from "prop-types";
import React from "react";
import { Link } from "react-router-dom";
import styled from "styled-components";
import { Badge, BoxGroupSection } from "../../base";

const StyledIndexOTU = styled(BoxGroupSection)`
    display: flex;
    justify-content: space-between;
`;

export const IndexOTU = ({ refId, changeCount, id, name }) => (
    <StyledIndexOTU>
        <Link to={`/refs/${refId}/otus/${id}`}>{name}</Link>
        <Badge>
            {changeCount} {`change${changeCount > 1 ? "s" : ""}`}
        </Badge>
    </StyledIndexOTU>
);

IndexOTU.propTypes = {
    refId: PropTypes.string.isRequired,
    changeCount: PropTypes.number.isRequired,
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
};

export default IndexOTU;
