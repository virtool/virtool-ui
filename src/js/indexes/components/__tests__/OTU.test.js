import { IndexOTU } from "../OTU";

describe("<IndexOTU />", () => {
    let props;

    beforeEach(() => {
        props = {
            refId: "foo",
            changeCount: 1,
            id: "bar",
            name: "baz"
        };
    });

    it("should render when [changeCount = 1]", () => {
        const wrapper = shallow(<IndexOTU {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
    it("should render when [changeCount = 2]", () => {
        props.changeCount = 2;
        const wrapper = shallow(<IndexOTU {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
