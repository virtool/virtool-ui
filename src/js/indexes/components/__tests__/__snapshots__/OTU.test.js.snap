// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IndexOTU /> should render when [changeCount = 1] 1`] = `
<OTU__StyledIndexOTU>
  <Link
    to="/refs/foo/otus/bar"
  >
    baz
  </Link>
  <Badge>
    1
     
    change
  </Badge>
</OTU__StyledIndexOTU>
`;

exports[`<IndexOTU /> should render when [changeCount = 2] 1`] = `
<OTU__StyledIndexOTU>
  <Link
    to="/refs/foo/otus/bar"
  >
    baz
  </Link>
  <Badge>
    2
     
    changes
  </Badge>
</OTU__StyledIndexOTU>
`;
